backend/qemu.pm

```diff
@@ -325,18 +325,20 @@ sub start_qemu {
         die "no dmi data for '$vars->{LAPTOP}'\n" unless -d "$bmwqemu::scriptdir/dmidata/$vars->{LAPTOP}";
     }
 
-    my $bootfrom = "";    # branch by "disk" or "cdrom", not "c" or "d"
+    my $bootfrom = '';    # branch by "disk" or "cdrom", not "c" or "d"
     if ($vars->{BOOT_HDD_IMAGE}) {
         # skip dvd boot menu and boot directly from hdd
         $vars->{BOOTFROM} //= 'c';
     }
-    if ($vars->{BOOTFROM} eq "d" || $vars->{BOOTFROM} eq "cdrom") {
-        $bootfrom = "cdrom";
-        $vars->{BOOTFROM} = "d";
-    }
-    if ($vars->{BOOTFROM} eq "c" || $vars->{BOOTFROM} eq "disk") {
-        $bootfrom = "disk";
-        $vars->{BOOTFROM} = "c";
+    if (my $bootfrom_var = $vars->{BOOTFROM}) {
+        if ($bootfrom_var eq 'd' || $bootfrom_var eq 'cdrom') {
+            $bootfrom = 'cdrom';
+            $vars->{BOOTFROM} = 'd';
+        }
+        if ($bootfrom_var eq 'c' || $bootfrom_var eq 'disk') {
+            $bootfrom = 'disk';
+            $vars->{BOOTFROM} = 'c';
+        }
```

Comment:
As you are already touching this, can you change second `if` to `elsif` and add `else` block throwing exception on unknown/unsupported boot order?